src/tools/pkcs11-tool.c

```diff
@@ -2695,7 +2698,7 @@ derive_key(CK_SLOT_ID slot, CK_SESSION_HANDLE session, CK_OBJECT_HANDLE key)
 		if (!ecpoint || !ecgroup)
 			util_fatal("Failed to parse other EC kry from %s", opt_input);
 
-		len = EC_POINT_point2oct(ecgroup, ecpoint, POINT_CONVERSION_UNCOMPRESSED, buf, sizeof(buf),NULL);
+		len = EC_POINT_point2oct(ecgroup, ecpoint, POINT_CONVERSION_UNCOMPRESSED, buf, buf_size, NULL);
```

Comment:
could you please detail the actual problem you are solving in the commit message?